### PR TITLE
Token id validation fix + refactor + tests

### PR DESCRIPTION
**Pull request by andrei-marinica**
## Pull request overview

This PR tightens ESDT token identifier validation to enforce a lowercase-hex random suffix, refactors the validation logic in `multiversx-chain-core`, and expands test coverage for edge cases (max ticker length, invalid chars, and invalid random suffix content).

**Changes:**
- Refactor `validate_token_identifier` into smaller helpers and return structured validation errors.
- Update the VM hook implementation to adapt to the new validation return type.
- Add/extend tests covering max-length tickers and stricter random-suffix validation (hex-only).

### CODE REVIEW

File: chain/core/src/token_identifier_util.rs [hunk 1]
```diff
@@ -8,52 +21,170 @@ pub const IDENTIFIER_MAX_LENGTH: usize = TICKER_MAX_LENGTH + ADDITIONAL_RANDOM_C
...
 
-    true
+    Ok(())
+}
+
+/// There is a VM implementation of this very function.
+/// This implementation is used for debugging mode and backwards compatibility.
+/// Using the VM implementation instead of this one saves ~0.6 kB of contract code.
+pub fn validate_token_identifier(token_id_slice: &[u8]) -> Result<(), TokenIdValidationError> {
+    let length = token_id_slice.len();
```
[the remark this answers is not in the document]
Comment (andrei-marinica) [reply chain/core/src/token_identifier_util.rs:66]: Will be in a minor release, it's ok.
